Linux compile fails (again) with NO_WEBVIEW set to ON

SLADE Version: current git
Operating System: Gentoo Linux

Issue Details:

This was fixed with Issue #730 for some days and then it was broken again :(

Compilation fails on Linux when NO_WEBVIEW is set to ON:

[ 64%] Building CXX object src/CMakeFiles/slade.dir/MainEditor/MainEditor.o
~/testuser/SLADE/src/MainEditor/MainEditor.cpp:98:43: error: 'void MainEditor::openDocs(string)' should have been declared inside 'MainEditor'
void MainEditor::openDocs(string page_name)
^
~/testuser/SLADE/src/MainEditor/MainEditor.cpp: In function 'void MainEditor::openDocs(string)':
/testuser/SLADE/src/MainEditor/MainEditor.cpp:100:15: error: 'class MainWindow' has no member named 'openDocs'
main_window->openDocs(page_name);

--- a/src/MainEditor/MainEditor.cpp
+++ b/src/MainEditor/MainEditor.cpp
@@ -94,9 +94,9 @@ EntryPanel* MainEditor::currentEntryPanel()
 {
 	return main_window->getArchiveManagerPanel()->currentArea();
 }
-+#ifdef USE_WEBVIEW_STARTPAGE
+#ifdef USE_WEBVIEW_STARTPAGE
 void MainEditor::openDocs(string page_name)
 {
 	main_window->openDocs(page_name);
 }
-+#endif
+#endif

--- a/src/MainEditor/MainEditor.cpp
+++ b/src/MainEditor/MainEditor.cpp
@@ -94,8 +94,9 @@ EntryPanel* MainEditor::currentEntryPanel()
 {
 	return main_window->getArchiveManagerPanel()->currentArea();
 }
-
++#ifdef USE_WEBVIEW_STARTPAGE
 void MainEditor::openDocs(string page_name)
 {
 	main_window->openDocs(page_name);
 }
++#endif

--- a/src/MainEditor/UI/MainWindow.cpp
+++ b/src/MainEditor/UI/MainWindow.cpp
@@ -921,7 +921,8 @@ void MainWindow::onHTMLLinkClicked(wxEvent& e)
 		string rs = href.Mid(9);
 		unsigned long index = 0;
 		rs.ToULong(&index);
-		SActionHandler::doAction("aman_recent", index);
+		SActionHandler::setWxIdOffset(index);
+		SActionHandler::doAction("aman_recent");
 		createStartPage();
 	}
 	else if (href.StartsWith("action://"))

